runtime/src/proofs.rs

```diff
@@ -66,12 +90,29 @@ fn validate_proof(matches: &mut Vec<[u8; 32]>, hash: [u8; 32], proofs: Vec<[u8;
     false
 }
 
+// prefill the required matches before proof validation
+fn pre_matches(static_proofs: [[u8; 32]; 3], doc_root: [u8; 32]) -> (bool, Vec<[u8; 32]>) {
+    let mut matches = Vec::new();
+    let data_root = static_proofs[0];
```

Comment:
Since the inner signing tree is static as well, I would rename:
- `data_Root` to `basic_data_root` or `left_data_root`
- `sibling_root` to `zk_data_root` or `right_data_root`
WDYT?